category_theory/functor.lean

```diff
@@ -49,21 +57,21 @@ namespace category
 variables (C : Type u₁) [𝒞 : category.{u₁ v₁} C]
 include 𝒞
 
-definition identity : C ↝ C := 
+protected definition identity : C ↝ C := 
 { obj      := λ X, X,
   map      := λ _ _ f, f,
   map_id   := begin /- `obviously'` says: -/ intros, refl end,
   map_comp := begin /- `obviously'` says: -/ intros, refl end }
 
 instance has_one : has_one (C ↝ C) :=
-{ one := identity C }
+{ one := category.identity C }
 
 variable {C}
 
-@[simp] lemma identity_to_has_one : (identity C) = 1 := rfl
+@[simp] protected lemma identity_to_has_one : (category.identity C) = 1 := rfl
 
-@[simp] lemma has_one.on_objects (X : C) : (1 : C ↝ C) X = X := rfl
-@[simp] lemma has_one.on_morphisms {X Y : C} (f : X ⟶ Y) : (1 : C ↝ C).map f = f := rfl
+@[simp] protected lemma has_one.on_objects (X : C) : (1 : C ↝ C) X = X := rfl
+@[simp] protected lemma has_one.on_morphisms {X Y : C} (f : X ⟶ Y) : (1 : C ↝ C).map f = f := rfl
```

Comment:
these don't all need to be protected, just `id` itself (assuming you call it `id` ;) )

Reply:
So we don't have the option of calling it `id`, as there's already a structure field `category.id`, for the identity morphism on an object. This `identity` is for the identity functor on the category itself.